django/contrib/auth/models.py

```diff
@@ -304,6 +304,8 @@ def has_perms(self, perm_list, obj=None):
         Return True if the user has each of the specified permissions. If
         object is passed, check if the user has all required perms for it.
         """
+        if isinstance(perm_list, str):
+            raise TypeError("has_perms() expects a list of strings, not a string. Are you expecting to use has_perm() instead?")
```

Comment:
I would raise a `ValueError`:
```suggestion
        if not isinstance(perm_list, (list, tuple)):
            ValueError('perm_list must be a list or tuple.')
```